contrib/platform/src/com/sun/jna/platform/win32/Tlhelp32.java

```diff
@@ -156,4 +159,109 @@ protected List getFieldOrder() {
             return Arrays.asList(new String[] { "dwSize", "cntUsage", "th32ProcessID", "th32DefaultHeapID", "th32ModuleID", "cntThreads", "th32ParentProcessID", "pcPriClassBase", "dwFlags", "szExeFile" });
         }
     }
+    
+
+    /**
+     * Describes an entry from a list of the modules belonging to the specified
+     * process.
+     * 
+     * @see <a href="https://msdn.microsoft.com/en-us/library/windows/desktop/ms684225(v=vs.85).aspx">MSDN</a>
+     */
+    public class MODULEENTRY32 extends Structure {
+    
+        /**
+         * A representation of a MODULEENTRY32 structure as a reference
+         */
+        public static class ByReference extends MODULEENTRY32 implements Structure.ByReference {
+            public ByReference() {
+            }
+    
+            public ByReference(Pointer memory) {
+                super(memory);
+            }
+        }
+    
+        public MODULEENTRY32() {
+            dwSize = new WinDef.DWORD(size());
+        }
+    
+        public MODULEENTRY32(Pointer memory) {
+            super(memory);
+            read();
+        }
+    
+        /**
+         * The size of the structure, in bytes. Before calling the Module32First
+         * function, set this member to sizeof(MODULEENTRY32). If you do not
+         * initialize dwSize, Module32First fails.
+         */
+        public DWORD dwSize;
+    
+        /**
+         * This member is no longer used, and is always set to one.
+         */
+        public DWORD th32ModuleID;
+    
+        /**
+         * The identifier of the process whose modules are to be examined.
+         */
+        public DWORD th32ProcessID;
+    
+        /**
+         * The load count of the module, which is not generally meaningful, and
+         * usually equal to 0xFFFF.
+         */
+        public DWORD GlblcntUsage;
+    
+        /**
+         * The load count of the module (same as GlblcntUsage), which is not
+         * generally meaningful, and usually equal to 0xFFFF.
+         */
+        public DWORD ProccntUsage;
+    
+        /**
+         * The base address of the module in the context of the owning process.
+         */
+        public Pointer modBaseAddr;
+    
+        /**
+         * The size of the module, in bytes.
+         */
+        public DWORD modBaseSize;
+    
+        /**
+         * A handle to the module in the context of the owning process.
+         */
+        public HMODULE hModule;
+    
+        /**
+         * The module name.
+         */
+        public char[] szModule = new char[MAX_MODULE_NAME32 + 1];
+    
+        /**
+         * The module path.
+         */
+        public char[] szExePath = new char[Kernel32.MAX_PATH];
```

Comment:
Reviving a couple of comments that got hidden by an outdated diff:

@twall suggested this in a now hidden comment:
> Because this depends on a `Structure` with explicit native wide strings, you should explicitly map to `Module32FirstW`. I believe you should still be able to reference with `Module32First` in client code.

I see the `com.sun.jna.FunctionMapper` interface  but I'm unsure how I could use it surgically (without potentially impacting the rest of `Kernel32`).

Is there something I should mark up `Module32First` and `Module32Next` with (e.g. an annotation)? I'd hate to break something else in `Kernel32`.

Last but not least, note that the Unicode functions do end in `-W` but the ANSI versions do *not* end in `-A` according to MSDN.

-----
Duplicating what I have here and adding a `W` to the end where appropriate is probably the easiest, quickest, and least likely to have a weird side effect. 

@dblock and @twall - I await your decision.